### PR TITLE
[Widgets] ConversationalWidget: homogenize options + bug fixes

### PR DESCRIPTION
**Pull request by SBrandeis**
# TL;DR

Fix several bugs and homogenize behavior with other widgets

### CODE REVIEW

File: packages/widgets/src/lib/components/InferenceWidget/widgets/ConversationalWidget/ConversationalWidget.svelte [hunk 1]
```diff
@@ -151,6 +150,15 @@
 		text = "";
 		error = "";
 		try {
+			const opts = {
+				dont_load_model: isOnLoadCall,
+				includeCredentials,
+				signal: abort?.signal,
+				use_cache: useCache,
+				wait_for_model: withModelLoading,
+			} satisfies Options;
```
Comment (SBrandeis) [inline packages/widgets/src/lib/components/InferenceWidget/widgets/ConversationalWidget/ConversationalWidget.svelte:159]: Aligns with what other widgets do

File: packages/widgets/src/lib/components/InferenceWidget/widgets/ConversationalWidget/ConversationalWidget.svelte [hunk 2]
```diff
@@ -151,6 +150,15 @@
...
+
+			tgiSupportedModels = await getTgiSupportedModels(apiUrl);
```
Comment (Wauplin) [inline packages/widgets/src/lib/components/InferenceWidget/widgets/ConversationalWidget/ConversationalWidget.svelte:161]: why removing this from `onMount`? I would assume the list never change.

Comment (SBrandeis) [reply packages/widgets/src/lib/components/InferenceWidget/widgets/ConversationalWidget/ConversationalWidget.svelte:161]: The main reason is that there is a race condition between `getTgiSupportedModels` and `applyWidgetExamples` on load.

The latter is called `onMount` by `WidgetExample`:
https://github.com/huggingface/huggingface.js/blob/d9c933a4611ddd55722a4eb39cbd95eb66ec205f/packages/widgets/src/lib/components/InferenceWidget/shared/WidgetExamples/WidgetExamples.svelte#L94-L117

Which would result in `ConversationalWidget.getOutput` being called before `getTgiSupportedModels` has completed. As a consequence, the example showed on load is not using the TGI streaming API although it should.


The change introduced in this PR makes sure `getTgiSupportedModels` is complete before proceeding with the inference call.
Note that calling `getTgiSupportedModels` several times in a row does not trigger more than one network call to fetch the info from the inference API.

------

>  I would assume the list never change.
This is mostly true during the inference widget's lifespan, but not absolutely true. The list represents models that are currently deployed with TGI in the inference API and can change over time.

Comment (Wauplin) [reply packages/widgets/src/lib/components/InferenceWidget/widgets/ConversationalWidget/ConversationalWidget.svelte:161]: > The change introduced in this PR makes sure getTgiSupportedModels is complete before proceeding with the inference call. 

Thanks for the explanation!

> Note that calling getTgiSupportedModels several times in a row does not trigger more than one network call to fetch the info from the inference API.

Oh great! Then it's all good :+1: 

> This is mostly true during the inference widget's lifespan, but not absolutely true. 

Yes indeed, I was talking about within the inference widget lifespan as well :+1:

**Review by Wauplin — APPROVED** [review]
(approval but wait for real reviewers :innocent:)

**Review by mishig25 — APPROVED** [review]
lgtm !

**Comment by mishig25** [comment]
unrelated idea: we should probably rename  `ConversationalWidget` -> `ChatWidget`

File: packages/widgets/src/lib/components/InferenceWidget/widgets/ConversationalWidget/ConversationalWidget.svelte [hunk 3]
```diff
@@ -13,7 +13,7 @@
 		AddedToken,
 	} from "@huggingface/tasks";
 	import { SPECIAL_TOKENS_ATTRIBUTES } from "@huggingface/tasks";
-	import { HfInference } from "@huggingface/inference";
+	import { HfInference, type Options } from "@huggingface/inference";
```
Comment (xenova) [inline packages/widgets/src/lib/components/InferenceWidget/widgets/ConversationalWidget/ConversationalWidget.svelte:16]:
```suggestion
	import { HfInference } from "@huggingface/inference";
	import type { Options } from "@huggingface/inference";
```
unless this is intended?

**Comment by xenova** [comment]
Can we merge this after https://github.com/huggingface/huggingface.js/pull/536 (so that the next version will also ensure 100% coverage of templates on the HF Hub?)

**Pull request merged**